aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -246,5 +247,37 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool || result.scalar_type() == kByte,
+      "The output tensor of lt must be a bool or byte tensor.");
+  auto iter = TensorIterator::comparison_op(result, self, other, true);
+  lt_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor lt(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  native::lt_out(result, self, other);
+  return result;
+}
+
+Tensor& lt_(Tensor& self, const Tensor& other) {
+  auto iter = TensorIterator::comparison_op(self, self, other, true);
+  lt_stub(iter.device_type(), iter);
+  return self;
+}
+
+Tensor lt(const Tensor& self, Scalar other) {
+  return native::lt(self, wrapped_scalar_tensor(other));
+}
+
+Tensor& lt_(Tensor& self, Scalar other) {
+  return native::lt_(self, wrapped_scalar_tensor(other));
+}
+
+Tensor& lt_out(Tensor& result, const Tensor& self, Scalar other) {
+  return native::lt_out(result, self, wrapped_scalar_tensor(other));
+}
```

Comment:
what's our plan for the rest of the operators? it would be a bit odd if we merge this to a state where `<` promotes its operands and other operators don't but if we're stacking these I guess we can land all stacks together. 

I don't love macros but it could also be error prone to copy/paste all this for each of `<, >, <=, >=, !=, ==`.

Reply:
I planned to land this first, and then work on PR for other ops so will try to find a solution to avoid duplications.

Reply:
We should also avoid inconsistent behavior though. On master, `torch.tensor(1) < 1.5` is False, and this would fix that issue but only specifically for the `<` operator. 

If we have differing behavior for different comparison operators it leaves us in a state that's difficult to explain. E.g. if we merge just `<` then `torch.tensor(1) < 1.5` will equal True but `1.5 > torch.tensor(1)` will still equal False. I think that inconsistency would be worse than what we currently have on master.

Reply:
The branch cut is soon and @nairbv brings up a good point about consistency since we are adding some type promotion behavior. @ifedan I think you should have a plan for if you want all of the comparison ops to go into the release, or if you want them to go into PyTorch after the release.